src/hooks/screen/useChatScreenHook.ts

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import 'react-native-get-random-values';
+
+import * as React from 'react';
+import SimpleToast from 'react-native-simple-toast';
+import {v4 as uuid} from 'uuid';
+
+import AnonymousMessageRepo from '../../service/repo/anonymousMessageRepo';
+import ChatSchema from '../../database/schema/ChatSchema';
+import SignedMessageRepo from '../../service/repo/signedMessageRepo';
+import UseChatScreenHook from '../../../types/hooks/screens/useChatScreenHook.types';
+import useChatUtilsHook from '../core/chat/useChatUtilsHook';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import {
+  CHANNEL_TYPE_ANONYMOUS,
+  CHANNEL_TYPE_GROUP,
+  CHANNEL_TYPE_PERSONAL
+} from '../../utils/constants';
+import {getAnonymousUserId, getUserId} from '../../utils/users';
+import {randomString} from '../../utils/string/StringUtils';
+
+function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
+  const {localDb, chat, refresh} = useLocalDatabaseHook();
+  const {selectedChannel, goBackFromChatScreen, goToChatInfoScreen} = useChatUtilsHook();
+
+  const [chats, setChats] = React.useState<ChatSchema[]>([]);
+  const {anon_user_info_emoji_name} = selectedChannel?.rawJson?.channel || {};
+  const initChatData = async () => {
+    if (!localDb && !selectedChannel) return;
+    try {
+      const myUserId = await getUserId();
+      const myAnonymousId = await getAnonymousUserId();
+      const data = (await ChatSchema.getAll(
+        localDb,
+        selectedChannel?.id,
+        myUserId,
+        myAnonymousId
+      )) as ChatSchema[];
+      setChats(data);
+    } catch (e) {
+      console.log(e, 'error get all chat');
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling could be improved. Currently, the error is just logged to the console. Consider adding better error handling or at least provide a user-friendly message.
```diff
- console.log(e, 'error get all chat');
+ SimpleToast.show("An error occurred while fetching chat data.");
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->